services/powchain/src/main/java/tech/pegasys/teku/services/executionengine/ExecutionEngineChannelImpl.java

```diff
@@ -0,0 +1,147 @@
+import tech.pegasys.teku.spec.executionengine.ExecutionEngineChannel;
+import tech.pegasys.teku.spec.executionengine.ForkChoiceState;
+import tech.pegasys.teku.spec.executionengine.PayloadAttributes;
+import tech.pegasys.teku.spec.schemas.SchemaDefinitionsMerge;
+import tech.pegasys.teku.ssz.type.Bytes8;
+
+public class ExecutionEngineChannelImpl implements ExecutionEngineChannel {
+  private static final Logger LOG = LogManager.getLogger();
+
+  private final ExecutionEngineClient executionEngineClient;
+  private final Spec spec;
+
+  public static ExecutionEngineChannelImpl create(String eeEndpoint, Spec spec) {
+    checkNotNull(eeEndpoint);
+    return new ExecutionEngineChannelImpl(new Web3JExecutionEngineClient(eeEndpoint), spec);
+  }
+
+  public ExecutionEngineChannelImpl(ExecutionEngineClient executionEngineClient, Spec spec) {
+    this.spec = spec;
+    this.executionEngineClient = executionEngineClient;
+  }
+
+  private static <K> K unwrapResponseOrThrow(Response<K> response) {
+    checkArgument(
+        response.getErrorMessage() == null,
+        "Invalid remote response: %s",
+        response.getErrorMessage());
+    return response.getPayload();
+  }
+
+  @Override
+  public SafeFuture<Optional<PowBlock>> getPowBlock(final Bytes32 blockHash) {
+    LOG.trace("calling getPowBlock(blockHash={})", blockHash);
+
+    return executionEngineClient
+        .getPowBlock(blockHash)
+        .thenPeek(powBlock -> LOG.trace("getPowBlock(blockHash={}) -> {}", blockHash, powBlock));
+  }
+
+  @Override
+  public SafeFuture<PowBlock> getPowChainHead() {
+    LOG.trace("calling getPowChainHead()");
+
+    return executionEngineClient
+        .getPowChainHead()
+        .thenPeek(powBlock -> LOG.trace("getPowChainHead() -> {}", powBlock));
+  }
+
+  @Override
+  public SafeFuture<tech.pegasys.teku.spec.executionengine.ForkChoiceUpdatedResult>
+      forkChoiceUpdated(
+          final ForkChoiceState forkChoiceState,
+          final Optional<PayloadAttributes> payloadAttributes) {
+
+    LOG.trace(
+        "calling forkChoiceUpdated(forkChoiceState={}, payloadAttributes={})",
+        forkChoiceState,
+        payloadAttributes);
+
+    return executionEngineClient
+        .forkChoiceUpdated(
+            ForkChoiceStateV1.fromInternalForkChoiceState(forkChoiceState),
+            PayloadAttributesV1.fromInternalForkChoiceState(payloadAttributes))
+        .thenApply(ExecutionEngineChannelImpl::unwrapResponseOrThrow)
+        .thenApply(ForkChoiceUpdatedResult::asInternalExecutionPayload)
+        .thenPeek(
+            forkChoiceUpdatedResult ->
+                LOG.trace(
+                    "forkChoiceUpdated(forkChoiceState={}, payloadAttributes={}) -> {}",
+                    forkChoiceState,
+                    payloadAttributes,
+                    forkChoiceUpdatedResult));
+  }
+
+  @Override
+  public SafeFuture<ExecutionPayload> getPayload(final Bytes8 payloadId, final UInt64 slot) {
```

Comment:
@ajsutton I changed the interface and the implementation here to lookup the schema